Userland/Libraries/LibGUI/TextEditor.h

```diff
@@ -192,6 +192,16 @@ class TextEditor
 
     void delete_text_range(TextRange);
 
+    // ^TextDocument::Client
```

Comment:
Your intuition is right that this is not the correct approach :)

I think a good fix here could be to create a `MoveLineUpOrDownCommand` object that inherits from `TextDocumentUndoCommand`, like we already have `InsertTextCommand` and `RemoveTextCommand`.